dali/operators/decoder/video/video_decoder_cpu.cc

```diff
@@ -12,66 +12,87 @@
-    thread_pool.AddWork([this, s, &output](int tid) {
-      DecodeSample(output[s], s);
-    }, volume(input[s].shape()));
-  }
-  thread_pool.RunAll();
-}
-
 DALI_SCHEMA(experimental__decoders__Video)
     .DocStr(
-        R"code(Decodes a video file from a memory buffer (e.g. provided by external source).
+        R"code(Decodes video files from memory buffers into sequences of frames.
+
+The operator accepts video files in common container formats (e.g. MP4, AVI). For CPU backend,
+FFmpeg is used for decoding. For Mixed backend, NVIDIA's Video Codec SDK (NVDEC) is used.
+
+Each output sample is a sequence of frames with shape (F, H, W, C) where:
+- F is the number of frames in the sequence (can vary between samples)
+- H is the frame height in pixels
+- W is the frame width in pixels 
+- C is the number of color channels
+
+.. code-block:: python
+
+    video_decoder = dali.experimental.decoders.Video(
+        frames=[0, 10, 20, 30, 40, 50, 40, 30, 20, 10, 0]
+    )
+
+Example 2: Extract a sequence of frames:
+
+.. code-block:: python
 
-The video streams can be in most of the container file formats. FFmpeg is used to parse video
- containers and returns a batch of sequences of frames with shape (F, H, W, C) where F is the
- number of frames in a sequence and can differ for each sample.)code")
+    video_decoder = dali.experimental.decoders.Video(
+        start_frame=0, sequence_length=10, stride=2
+    )
+
+Example 3: Pad the sequence by repeating the last frame:
+
+.. code-block:: python
+
+    video_decoder = dali.experimental.decoders.Video(
+        start_frame=0, sequence_length=100, stride=2, pad_mode="edge"
+    )
+
+)code")
     .NumInput(1)
     .NumOutput(1)
-    .InputDox(0, "buffer", "TensorList", "Data buffer with a loaded video file.")
+    .InputDox(0, "encoded", "TensorList", "Encoded video stream")
     .AddOptionalArg("affine",
-    R"code(Applies only to the mixed backend type.
+    R"code(Whether to pin threads to CPU cores (mixed backend only).
+
+If True, each thread in the internal thread pool will be pinned to a specific CPU core.
+If False, threads can migrate between cores based on OS scheduling.)code", true)
+    .AddOptionalArg<std::vector<int>>("frames",
+    R"code(Specifies which frames to extract from each video by their indices.
+
+The indices can be provided in any order and can include duplicates. For example, [0,10,5,10] would extract:
+- Frame 0 (first frame)
+- Frame 10
+- Frame 5 
+- Frame 10 (again)
+
+This argument cannot be used together with ``start_frame``, ``sequence_length``, ``stride`` or ``pad_mode``.)code", nullptr, true)
+    .AddOptionalArg<int>("start_frame",
+    R"code(Index of the first frame to extract from each video. Cannot be used together with frames argument.)code", nullptr, true)
+    .AddOptionalArg<int>("stride",
+    R"code(Number of frames to skip between each extracted frame. Cannot be used together with ``frames`` argument.)code", nullptr, true)
```

Comment:
One of these has the `frames` rendered in monospace and the other in regular font. Would you consider unifying it?